### PR TITLE
Add error checking for cpp examples.

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##
In current cpp example, if data file isn't exist, example application will crash with sig 11 .
Add error checking to avoid crash.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: cpp-package/example/mlp_cpu.cpp [hunk 1]
```diff
@@ -63,10 +63,14 @@ int main(int argc, char** argv) {
                                         };
 
   auto train_iter =  MXDataIter("MNISTIter");
-  setDataIter(&train_iter, "Train", data_files, batch_size);
+  if (!setDataIter(&train_iter, "Train", data_files, batch_size)) {
+
```
Comment (TaoLv) [inline cpp-package/example/mlp_cpu.cpp]: missing return value?

File: cpp-package/example/alexnet.cpp [hunk 2]
```diff
@@ -244,10 +244,14 @@ int main(int argc, char const *argv[]) {
                               };
 
   auto train_iter =  MXDataIter("MNISTIter");
-  setDataIter(&train_iter, "Train", data_files, batch_size);
+  if (!setDataIter(&train_iter, "Train", data_files, batch_size)) {
+    return 1;
```
Comment (TaoLv) [inline cpp-package/example/alexnet.cpp]: Can we have some descriptive error messages?

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by lanking520** [comment]
Restart the test

**Pull request merged**